src/lib.rs

```diff
@@ -286,30 +281,33 @@ impl Default for RequestParams {
 }
 
 /// Get a default `Client` and `RequestParams`.
-pub fn default() -> (hyper::Client, RequestParams) {
-    (hyper::Client::new(), RequestParams::default())
+pub fn default() -> (reqwest::Client, RequestParams) {
+    //FIXME: Bad in libs
+    let client = reqwest::Client::new().unwrap();
+    (client, RequestParams::default())
 }
 
 macro_rules! req_with_body {
     ($client:ident, $url:ident, $req:ident, $params:ident, $method:ident) => ({
     	let body = $req.body.expect("Expected this request to have a body. This is a bug, please file an issue on GitHub.");
 
 		let body: &[u8] = (*body).borrow();
-		let mut cursor = Cursor::new(body);
+//FIXME: My current understanding of copying, etc. doesn't get me far enough to grok this
```

Comment:
So I had a quick poke through `reqwest` and it looks like this isn't really a thing there. It takes a `Vec<u8>`, `String`, `&[u8]` or `&str` and then takes a copy of it anyway. See [here](https://github.com/seanmonstar/reqwest/blob/master/src/body.rs#L77-L89).

Instead, `hyper` can take a generic `Read`, and use that for the body. See [here](https://github.com/hyperium/hyper/blob/0.9.x/src/client/mod.rs#L424-L429). So you can create an owned [`Cursor`](https://doc.rust-lang.org/stable/std/io/struct.Cursor.html) over a borrowed slice of bytes and it avoids creating a copy of the data for buffering.

I might log an issue in `reqwest` about supporting zero-copy body buffers. So for the purposes of this library, we can revert it to the way you had it originally, with a comment that it's going to copy the body. At least if we're copying it in our code it's obvious.

Sorry for misleading you!